app/BaseHistoryExporter.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace App\Exporters;
+
+use App\Repositories\HistoryRepository;
+
+abstract  class BaseHistoryExporter
+{
+    protected $historyRepository;
+
+    public function __construct()
+    {
+        $this->historyRepository = new HistoryRepository();
+    }
+
+    public function export($date = null)
+    {
+        return (is_null($date))
+            ? $this->exportAll()
+            : $this->exportByDate($date);
+    }
+
+    public function exportByDate($date)
+    {
+        $data = $this->historyRepository->get(['date' => $date]);
+
+        return $this->showHistoryItems([$date => $data]);
+    }
+
+    public function exportAll()
+    {
+        return $this->showHistoryItems($this->historyRepository->allGroupedBy('date'));
+    }
+
+    protected function writeHistoryLine($historyItem)
+    {
+        $isBasicMathOperation = in_array($historyItem['sign'], BASIC_COMMANDS);
+        $prefix = ($isBasicMathOperation) ? '   ' : '(!)';
+
+        return "{$prefix} {$historyItem['first_operand']} {$historyItem['sign']} {$historyItem['second_operand']} = {$historyItem['result']}";
+    }
+
+    protected function showHistoryItems($data)
+    {
+        foreach ($data as $date => $historyItems) {
+            $this->output("{$date}: ");
+
+            foreach ($historyItems as $historyItem) {
+                $historyLine = $this->writeHistoryLine($historyItem);
+
+                $this->output($historyLine);
+            }
+        }
+
+        return write_symbol_line(15, '=');
```

Comment:
exporter class shouldn't make console output, if you want to draw this line inside the console - just redefine  `showHistoryItems` inside the HistoryConsoleExporter class

```php
protected function showHistoryItems($data)
{
    parent:: showHistoryItems($data);

    return write_symbol_line(15, '=');
}
```